### PR TITLE
#282 Edit Driver in back office

### PR DESCRIPTION
**Pull request by IsraelKariti**
1. create a router + controller + api call for the updating of the data in the db
2. create the modal + info components for updating the driver
3. add "Edit" button to table

### CODE REVIEW

**Comment by iddanh** [comment]
@IsraelKariti This PR should be to `main` branch I think

**Comment by IsraelKariti** [comment]
ניסיתי עכשיו, ונוצרו יותר מדי קונפליקטים עם קוד שאני לא מכיר, מעדיף שזה ישאר בopen to all

**Review by iddanh — CHANGES_REQUESTED** [review]
@IsraelKariti Tried it locally and got an error when submitting:
image

Looks like you forgot to update the `openapi.yaml` file with the new updateDriver endpoint

**Comment by IsraelKariti** [comment]
It seems that .gitignore specify the api-client directory since the api-client is a generated code.
Therefore I can't include the changes i have made to DriverApi.ts which is inside the api-client directory (namely the updateDriver function)
1. Should we changed .gitignore?
2. Should the updateDriver be generated automatically somehow?
3. Should the .yaml be updated automatically as well?

**Comment by iddanh** [comment]
This code is indeed autogenerated and should not be committed. It is generated based on the code in the yaml file, so you should make your updates there and commit them

**Comment by IsraelKariti** [comment]
@iddanh First time i encounter autogeneration, very cool concept, thank you, I have now updated the .yaml file

File: client/src/Backoffice/components/Main/components/modals/EditDriverModal/EditDriverCarInfo/EditDriverCarInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    register,
+    setValue,
+    formState: { errors }
+  } = useFormContext<Driver>();
+
+  useEffect(() => {
+    setValue('firstName', driver.firstName || '');
+    setValue('lastName', driver.lastName || '');
+    setValue('nationalId', driver.nationalId || '');
+  }, [driver, setValue]);
```
Comment (iddanh) [inline client/src/Backoffice/components/Main/components/modals/EditDriverModal/EditDriverCarInfo/EditDriverCarInfo.tsx:18]: When I tried it, modifying these values did not work. Whey did you need this useEffect?

**Review by iddanh — CHANGES_REQUESTED** [review]
Much better, mostly works now except for one bug

**Comment by IsraelKariti** [comment]
@iddanh You are right, I have now deleted useEffect. It was a leftovers from some previous trial and error

**Review by iddanh — APPROVED** [review]

**Comment by iddanh** [comment]
@IsraelKariti we probably want these changes also in `main` branch. Can you do a new PR and cherry pick these commits?

**Pull request merged**